presto-main/src/main/java/io/prestosql/sql/analyzer/Analysis.java

```diff
@@ -783,6 +787,7 @@ public void addColumnMask(Table table, String column, Expression mask)
                     NodeRef<Table> table = entry.getKey();
 
                     List<ColumnInfo> columns = referencedFields.get(table).stream()
+                            .filter(field -> field.getName().isPresent()) // For DELETE queries, the synthetic column for row id doesn't have a name
```

Comment:
Should we add a check afterwards that the count matches unless it is a delete query?

Reply:
No, I don't think so. This piece of code doesn't particularly care about whether it's a DELETE. It should work generically for any case where fields without a name are synthesized. The comment is there to clarify why that might happen.